src/components/AppShelf/AddAppShelfItem.tsx

```diff
@@ -407,6 +407,42 @@ export function AddAppShelfItemForm(props: AddAppShelfItemFormProps) {
                   />
                 </>
               )}
+              {form.values.type === 'NZBGet' && (
+                <>
+                  <TextInput
+                    label={t('modal.tabs.options.form.integrations.nzbget.username.label')}
+                    placeholder={t(
+                      'modal.tabs.options.form.integrations.nzbget.username.placeholder'
+                    )}
+                    value={form.values.username}
+                    onChange={(event) => {
+                      form.setFieldValue('username', event.currentTarget.value);
+                    }}
+                    error={
```

Comment:
To be honest, I just copied the code from the [Trasmission block](https://github.com/ajnart/homarr/blob/master/src/components/AppShelf/AddAppShelfItem.tsx#L374). It looks like your same question could be applied there too. So I guess the question now is, should I fix it? I could check to make sure `username` and `password` are not null/empty in the [form validation](https://github.com/ajnart/homarr/blob/master/src/components/AppShelf/AddAppShelfItem.tsx#L127).

Reply:
I think that we shouldn't restrict the user. Are null / empty values invalid or impossible to use with NZBGet? If not, I think we shouldn't restrict it. Can we resolve this issue or is there still work left?

Reply:
I think I have taken care of this now. If the user enables the Usenet module, the newly added [NzbgetClient](https://github.com/jonjon1123/homarr/blob/282-nzbget-download-client/src/pages/api/modules/usenet/nzbget/nzbget-client.ts#L4) will check to make sure valid config is provided.

Reply:
@jonjon1123 
sorry just found this after my comment. To check for null just add "required" to the fields classes. I removed it because some users had no user/pass and were complaining (although all logins worked)

Reply:
Also ideally we'd move translations to something more "common" like password and username is not usenet specific. Perhaps it could be moved above in the translation tree

Reply:
Technically it isn't required to provide a username & password to use NZBGet with Homarr. It is only required if you are going to enable to Usenet module. That's why I added the check to the new [NzbgetClient](https://github.com/jonjon1123/homarr/blob/282-nzbget-download-client/src/pages/api/modules/usenet/nzbget/nzbget-client.ts#L4) that only gets invoked as part of the Usenet module.